sdk/openai/Azure.AI.OpenAI/src/Custom/AzureCognitiveSearchChatExtensionConfiguration.cs

```diff
@@ -71,6 +71,29 @@ public AzureCognitiveSearchChatExtensionConfiguration(AzureChatExtensionType typ
             IndexName = indexName;
         }
 
+        // CUSTOM CODE NOTE: We keep this constructor as internal and users must set the search key via the SetSearchKey method instead.
+
+        /// <summary> Initializes a new instance of AzureCognitiveSearchChatExtensionConfiguration. </summary>
+        /// <param name="type">
+        /// The type label to use when configuring Azure OpenAI chat extensions. This should typically not be changed from its
+        /// default value for Azure Cognitive Search.
+        /// </param>
+        /// <param name="searchEndpoint"> The absolute endpoint path for the Azure Cognitive Search resource to use. </param>
+        /// <param name="searchKey"> The API admin key to use with the specified Azure Cognitive Search endpoint. </param>
+        /// <param name="indexName"> The name of the index to use as available in the referenced Azure Cognitive Search resource. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="searchEndpoint"/>, <paramref name="searchKey"/> or <paramref name="indexName"/> is null. </exception>
+        internal AzureCognitiveSearchChatExtensionConfiguration(AzureChatExtensionType type, Uri searchEndpoint, string searchKey, string indexName)
```

Comment:
All the others could be deleted from custom code, so that it could catch up with generated code.

Reply:
We still need to the public constructors in this file because we're doing something a little bit special. However, I was able to remove some of the properties and let them be generated.